### ISSUE TITLE
`tsc-strict` doesn't work with `--watch` flag

### ISSUE THREAD

**Issue by jandreola** [allegro/typescript-strict-plugin#79]
Documentation mentions that all `tsc` flags are supported by `tsc-strict`. But when I try to run `tsc-strict` with `--watch` I get the command stuck on `⠏ Looking for strict files...`.

Running the command in non watch mode works correctly. And the IDE also show errors correctly.

Any ideas if that is some sort of limitation of `tsc-strict` command?

I'm using `typescript-strict-plugin@2.4.4` and `typescript@5.4.5`

**Comment by KostkaBrukowa** [comment]
Yes, `tsc-strict` wasn't designed to work with `--watch` in mind and it wont work. It basically runs `tsc` twice, one time with `strict` flag and compares the outputs after `tsc` ends, and if you run with `--watch` flag the output will never be collected because `tsc` will never end

**Comment by heyimalex** [comment]
You could potentially create a PR to add this. There is an example of [an incremental program watcher](https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API#writing-an-incremental-program-watcher) inside of the typescript docs, and you'd just filter out errors coming from strict files.

[Here's](https://gist.github.com/heyimalex/74460a309d87df5f690f0b56022f43cd) what I rigged up, though it's not directly usable because it uses internal tools, maybe a good starting place. The `strictify` command is just generating a `tsconfig.strict.json` which is a copy of the base `tsconfig.json` but with strict enabled. There's definitely a way to handle that programatically.
